src/eval-frame/components/reps/rep-utils/tiny-rep-serializer.js

```diff
@@ -0,0 +1,80 @@
+export function getClass(obj) {
+  if (obj === null) return "Null";
+  if (obj === undefined) return "Undefined";
+  return obj.constructor.name;
+}
+
+export const getType = obj =>
+  Object()
+    .toString.call(obj)
+    .split(" ")[1]
+    .slice(0, -1);
+
+const typesWithLength = [
+  "Array",
+  "String",
+  "Int8Array",
+  "Uint8Array",
+  "Uint8ClampedArray",
+  "Int16Array",
+  "Uint16Array",
+  "Int32Array",
+  "Uint32Array",
+  "Float32Array",
+  "Float64Array",
+  "Function",
+  "GeneratorFunction"
+];
+
+const typesWithByteLength = ["ArrayBuffer", "DataView"];
+
+const typesWithSize = ["Map", "Set"];
+
+export function objSize(obj) {
+  if (obj === null) return undefined;
+  if (obj === undefined) return undefined;
+  const type = getType(obj);
+  if (type === "RegExp") return obj.source.length;
+  if (typesWithLength.includes(type)) return obj.length;
+  if (typesWithByteLength.includes(type)) return obj.byteLength;
+  if (typesWithSize.includes(type)) return obj.size;
+  return Object.getOwnPropertyNames(obj).length;
+}
+
+const MAX_TINY_STRING_LEN = 20;
+const TRUNCATION_LEN = 12;
+
+const truncateString = s =>
+  s.length > MAX_TINY_STRING_LEN
+    ? [s.slice(0, TRUNCATION_LEN), true]
+    : [s, false];
+
+export function tinyRepStringify(obj) {
+  const type = getType(obj);
+  let stringVal;
+  if (["Number", "Boolean", "Undefined", "Null", "Symbol"].includes(type)) {
+    stringVal = String(obj);
+  } else if (type === "Date") {
+    stringVal = obj.toISOString().slice(0, 19);
```

Comment:
What does this end up stripping off?  Time zone?

Reply:
time zone and milliseconds